pkg/kubectl/cmd/util/factory.go

```diff
@@ -1244,20 +1311,24 @@ func (f *factory) PrintObject(cmd *cobra.Command, mapper meta.RESTMapper, obj ru
 		return err
 	}
 
-	printer, err := f.PrinterForMapping(cmd, mapping, false)
+	printer, err := pmf.PrinterForMapping(cmd, mapping, false)
 	if err != nil {
 		return err
 	}
 	return printer.PrintObj(obj, out)
 }
 
-func (f *factory) PrinterForMapping(cmd *cobra.Command, mapping *meta.RESTMapping, withNamespace bool) (kubectl.ResourcePrinter, error) {
+func (f *factory) PrintObject(cmd *cobra.Command, mapper meta.RESTMapper, obj runtime.Object, out io.Writer) error {
+	return PrintObject(f, cmd, mapper, obj, out)
+}
+
+func PrinterForMapping(ccf ClientConfigFactory, pf PrinterFactory, cmd *cobra.Command, mapping *meta.RESTMapping, withNamespace bool) (kubectl.ResourcePrinter, error) {
 	printer, ok, err := PrinterForCommand(cmd)
 	if err != nil {
 		return nil, err
 	}
 	if ok {
-		clientConfig, err := f.ClientConfig()
+		clientConfig, err := ccf.ClientConfig()
```

Comment:
This is used to get an output version, which does nothing with the one version we have now, which we don't want to have in the future, and which looks like the only reason this method is large.

Is this the only reason you're doing this?